src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IConversionExpression.cs

```diff
@@ -0,0 +1,2991 @@
+﻿using System;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Semantics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    // Test list drawn from Microsoft.CodeAnalysis.CSharp.ConversionKind
+    public partial class IOperationTests : SemanticModelTestBase
```

Comment:
Hmm, I take that back, the compiler does seem to generate a BoundConversion with `ConversionKind.NoConversion` for the mismatched delegate type case as well. I'll dig a bit more.